### ISSUE TITLE
🐞 `dagger call --help` lists arguments as functions

### ISSUE THREAD

**Issue by MikaelElkiaer** [dagger/dagger#7455]
### What is the issue?

Fields in a module or custom type get listed as functions.
This is true at least for Go, have not tried others.

This is also true when using module constructors.

### Dagger version

dagger v0.11.4 (registry.dagger.io/engine) linux/amd64

### Steps to reproduce

I have a simple example here: https://github.com/MikaelElkiaer/daggerverse/blob/main/testing.go, from which I can run:

```bash
dagger --mod github.com/mikaelelkiaer/daggerverse call testing --help
```

### Log output

Output is:

```
USAGE
  dagger call testing <function>

FUNCTIONS
  interrupt     
  main          

INHERITED OPTIONS
  -d, --debug             show debug logs and full verbosity
      --json              Present result as JSON
  -m, --mod string        Path to dagger.json config file for the module or a directory containing that file. Either local path (e.g.
                          "/path/to/some/dir") or a github repo (e.g. "github.com/dagger/dagger/path/to/some/subdir")
  -o, --output string     Path in the host to save the result to
      --progress string   progress output format (auto, plain, tty) (default "auto")
  -s, --silent            disable terminal UI and progress output
  -v, --verbose count     increase verbosity (use -vv or -vvv for more)

Use "dagger call testing [command] --help" for more information about a command.
```

`main` is not a function, but a field on the custom type Struct.

**Comment by helderco** [comment]
Hi @MikaelElkiaer, that is expected, see [Develop with Go: Visibility](https://docs.dagger.io/manuals/developer/go/892021/visibility).

If you don't want to expose a function for a struct field, you can add the `// +private` pragma.

**Comment by MikaelElkiaer** [comment]
> Hi @MikaelElkiaer, that is expected, see [Develop with Go: Visibility](https://docs.dagger.io/manuals/developer/go/892021/visibility).
> 
> If you don't want to expose a function for a struct field, you can add the `// +private` pragma.

Thanks, totally missed that - too focused on searching this repo.

However, this raises 2 further questions:

1. What is the point of this feature - why would I want to expose my fields as functions? Particuilarily when looking at https://docs.dagger.io/manuals/developer/go/520657/constructor, there is an info box at the bottom stating I should have the fields as public, it seems to me that these use cases might be conflicting.
2. I do not understand the example in the doc - why is the `job` field not exposed? I get that `age` is not as it has the `// +private` pragma.

**Comment by MikaelElkiaer** [comment]
Final question on this.

For Go, why a private pragma, instead of public?
I get that Go enforces visibility based on casing, but is it not an explicit choice in the Go SDK to turn the public Go fields into Dagger functions?
To me it sounds like fields, even though public in Go, could be private by default in Dagger - which I understand would then be the same behavior for Python and TypeScript.

Thanks for your elaborations so far, keep up the good work!

**Comment by helderco** [comment]
> To me it sounds like fields, even though public in Go, could be private by default in Dagger - which I understand would then be the same behavior for Python and TypeScript.

That is true but at the time it was introduced, it was more consistent with the general behavior of the Go SDK to expose by default. Pragmas weren't around from the beginning and they were added mostly to control what arguments are optional or not (difficult because Go doesn't support default values). 

If you made that change today:
- It would be a breaking change for many people.
- Still inconsistent with functions that are exposed by default (unless lower cased).
- From the modules that I've seen, in any language, exposing fields seems to be more common than not.

**Comment by MikaelElkiaer** [comment]
I get it now, and I do see the usefulness.
I think my confusion arose from how I discovered it.

I had not looked at the section you linked about [Visibility](https://docs.dagger.io/manuals/developer/go/892021/visibility), but on the other hand I used [Module Constructor](https://docs.dagger.io/manuals/developer/go/520657/constructor/).
I then stumbled upon my fields turning up as functions, though I only added fields in order to use as constructor arguments.

I see how this behaviour could be useful for grabbing intermediate steps or state when using composite types.
However, it still does not make sense to me to expose module constructor arguments in this manner - for now I will turn them private.
